From 17bb756bfc7189d83d7a9dc0238080bd3ba29e11 Mon Sep 17 00:00:00 2001
From: Nagaraj S <sivasankaranx.nagaraj@intel.com>
Date: Fri, 25 Mar 2016 16:45:45 -0700
Subject: [PATCH 372/441] ce_mailbox:Init_completion move to static declaration

IssueID:   HSD# 245832
IssueType: Defect
Symptom:   Initialization should be during driver init.
           Should not Initialize each IOCTL call.
RootCause: Multiple Initialization may cause the kernel crash.
Resolution: Added static declaration and initialization.
Test:      Ran a test to verify ce_mailbox driver behavior.
Reviewer:  Warden Brett T
ReviewID:  CCID-133260

Change-Id: Id0923fff5d9d36c4bc7191b22879a35a2102abdf
---
 drivers/misc/ce_mailbox.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/drivers/misc/ce_mailbox.c
+++ b/drivers/misc/ce_mailbox.c
@@ -49,7 +49,6 @@ static struct proc_dir_entry   *mbx_proc
 static struct npcpu_appcpu_mbx_user    mbx_usr_info;
 struct npcpu_rpc_info rpc_info;
 bool glob_mbx_is_initialized = false;
-struct completion cmd_done;
 #define NPCPU_IP_START_BYTE 0
 #define APPCPU_IP_START_BYTE 4
 #define NETMASK_START_BYTE 8
@@ -57,6 +56,8 @@ struct completion cmd_done;
 #define MAX_NO_BYTE 4
 #define BYTE_SIZE 8
 
+static DECLARE_COMPLETION(cmd_done);
+
 /* delay macro for future use.*/
 //#define MBX_MAX_POLLING_DELAY          (msecs_to_jiffies(30000))
 
@@ -164,8 +165,7 @@ static int mbx_close(struct inode *inode
 /* Polloing on specivec callback from mailbox - max wait forever to get RPC info from ARM  */
 static int mbx_wait_till_ready()
 {
-        init_completion(&cmd_done);
-        wait_for_completion_interruptible(&cmd_done);
+	wait_for_completion(&cmd_done);
         return 0;
 }
 
